src/lib/core.d.ts

```diff
@@ -230,7 +232,8 @@ interface Function {
       * @param thisArg An object to which the this keyword can refer inside the new function.
       * @param argArray A list of arguments to be passed to the new function.
       */
-    bind(thisArg: any, ...argArray: any[]): any;
+    bind<T, U>(this: (this: T, ...argArray: any[]) => U, thisArg: T, ...argArray: any[]): (this: void, ...argArray: any[]) => U;
+    bind(this: Function, thisArg: any, ...argArray: any[]): any;

```

Comment:
1. I would like `this` to show up in the quick info. It's quite relevant to this method.
2. I want to say `this: Function` instead of `this: this` because it's a smaller type (thus, easier to call, harder to implement). I *hope* you can't monkey patch `bind` so implementation of `bind` shouldn't be an issue ... I'll go see what I can learn about that. It seems like something people might actually do in javascript.
3. `this: Function` is easier to read than `this: this` for first-time readers of `this` parameters.
3. Uniformity with the first overload.